src/guiguts/tools/pptxt.py

```diff
@@ -0,0 +1,2591 @@
+"""PPtxt tool"""
+
+
+from guiguts.checkers import CheckerDialog
+from guiguts.maintext import maintext
+from guiguts.utilities import IndexRowCol, IndexRange
+
+from typing import Dict, Sequence, List
+import regex as re
+
+########################################################
+# pptxt.py
+# Perl author: Roger Franks (DP:rfrank) - 2009
+# Go author: Roger Franks (DP:rfrank) - 2020
+# Python author: Quentin Campbell (DP:qgc) - 2024
+# Last edit: 7-mar-2024
+########################################################
+
+
+def pptxt() -> None:
```

Comment:
We really shouldn't just put everything in the file in one function -- there's no need and it serves no purpose.

It would be better to put functions and globals at the top and have a function that is the "main" that does the work.

Integration-wise, I would expect guiguts to have a function inside `application.py` (or similar) that is the glue between Guiguts (which knows about the text) and pptxt as the tool. `pptxt.py` should not know, or need to know, anything about CheckerDialog or maintext or any of the other GG-specific bits -- treat it as a library with non-GG-specific interfaces.

Reply:
Sorry for not communicating properly with everyone (especially @cpeel) about dropping the requirement for pptxt to be a standalone tool. I don't want to end up making extra work for everyone, so I'll liaise directly with @G4OEU about the additional changes he has already made that cover some of the points above, and agree any other tweaks before we get this first major tool merged in :tada:
From that point, we can have smaller changes in commits/PRs rather than the large iterations that Quentin has kindly hurdled over as I moved the goalposts.

Reply:
Ah, the integration of GG objects in the class makes sense in that framework 👍 . My only request is that we try and move the contents of the one function up a level so they are in the top-level scope.

Thanks for all your work on this @G4OEU! It will be great to have pptxt directly usable within GG2!

Reply:
Will do. Nigel has already asked for that in direct correspondence with me.

Thank you for your kind remarks.